crates/primitives/src/kzg/env_settings.rs

```diff
@@ -0,0 +1,30 @@
+use super::{
+    generated::{G1_POINTS, G2_POINTS},
+    KzgSettings,
+};
+use alloc::{boxed::Box, sync::Arc};
+use once_cell::race::OnceBox;
+
+/// KZG Settings.
+#[derive(Debug, Clone, Default, Eq, PartialEq)]
+pub enum EnvKzgSettings {
+    #[default]
+    Default,
+    Custom(Arc<c_kzg::KzgSettings>),
+}
```

Comment:
this needs proper docs

* used for point eval precompile
* Default: loaded from eth mainnet trusted setup
* Custom: use the given kzgsettings in the precompile